components/director/internal/domain/formation/notifications.go

```diff
@@ -268,12 +266,11 @@ func (ns *notificationsService) updateLastNotificationSentTimestamp(ctx context.
 
 	if fa != nil {
 		log.C(ctx).Infof("Updating the last notification sent timestamp for formation assignment with ID: %s", fa.ID)
-		fa.SetLastNotificationSentTimestamp(time.Now())
-		if err := ns.formationAssignmentRepo.Update(ctx, fa); err != nil {
+		if err := ns.formationAssignmentRepo.UpdateLastNotificationSentTimestamps(ctx, fa.ID); err != nil {
 			// That covers the case when we send two unassign notifications to one participant
 			// and the response of the first notification is returned and processed, which deletes the formation assignment,
 			// while the second notification still hasn't been sent.
-			if webhookNotificationReq.GetOperation() == model.UnassignFormation && (apperrors.IsNotFoundError(err) || apperrors.IsUnauthorizedError(err)) { // the not found error is disguised behind the unauthorized error in case of update
```

Comment:
Is the unauthorized not applicable anymore? It's still an update.

Reply:
I think even it might not return NotFound either, since it's not using the regular `updater`. But we should handle errors correctly, in order not to reintroduce race conditions again.

Reply:
About the first part of the sentence - we use SQL directly but if the error is `sql.ErrNoRows` in the `MapSQLError` func we can still receive a not found error. That's why I left the not found check.
Regarding the second part of the sentence - I didn't understand what we should do?

Reply:
My bad, I didn't realize that the MapSQLError returns a NotFound error in such a case. In that case it seems ok as is.